### PR TITLE
New consul-template backed maintenance page

### PR DESCRIPTION
**Pull request by perangel**

### CODE REVIEW

File: roles/haproxy/files/maintenance.tmpl [hunk 1]
```diff
@@ -0,0 +1,101 @@
+<!DOCTYPE html>
```
Comment (pmbauer) [inline roles/haproxy/files/maintenance.tmpl]: do we want to add :no-cache directive here?
see https://www.digitalocean.com/community/questions/haproxy-change-503-page-with-a-custom
```
HTTP/1.0 503 Service Unavailable
Cache-Control: no-cache
Connection: close
Content-Type: text/html
```

File: roles/haproxy/files/haproxy.cfg [hunk 2]
```diff
@@ -3,3 +3,9 @@ template {
   destination = "/haproxy/haproxy.cfg"
   command = "/usr/sbin/haproxy -D -p /var/run/haproxy.pid  -f /haproxy/haproxy.cfg -sf $(cat /var/run/haproxy.pid) || true"
 }
+
+template {
```
Comment (pmbauer) [inline roles/haproxy/files/haproxy.cfg]: does haproxy cache the pages in memory?

**Comment by pmbauer** [comment]
very nice; thanks for this

File: roles/haproxy/files/maintenance.tmpl (hunk 1 above)
Comment (perangel) [inline roles/haproxy/files/maintenance.tmpl]: Unfortunately while the browser does not cache the page, HAProxy appears to be caching it. I think it's safe to take the hit whenever the keys change, given that we'll seldom actually change them. It's just more of a plan-b option we have in our back pocket if something really catastrophic happens and we want to notify users with a note or something

Comment (pmbauer) [inline roles/haproxy/files/maintenance.tmpl]: lgtm then

**Pull request merged**